### PR TITLE
fix: tds adjusted totals for e-Invoice and e-Waybill

### LINKED ISSUE #763: Incorrect Taxable Amount in GSTR-1, e-Invoice, e-WayBill
### Information about bug

While selecting Nil Rated or Exempted Item (Bran, Suji, Maida, Atta, Whole Wheat,....)
If TCS Applicable then, it added tcs value in taxable value. which should not.
eg. In below image Taxable amount should be ₹ 5,35,691.52 but in e-WayBill, e-Invoice, GSTR-1 showing ₹ 5,36,227.212, which is incorrect.
plz look into this,
Thanks.
image

image

### Version

image

### Relevant log output / Stack trace / Full Error Message.

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by vorasmit**
- [x] Transaction Data (e-Waybill and e-Invoice)
- [x] GSTR-1
- [x] GSTR-3B

closes: #763

### CODE REVIEW

**Comment by Govind-Jangid** [comment]
@vorasmit Sir,
I tested above tds-adjusted-totals in this scenario tcs amount is perfect but taxable amount is still showing incorrect.
fyr
image

image

**Comment by vorasmit** [comment]
> @vorasmit Sir, I tested above tds-adjusted-totals in this scenario tcs amount is perfect but taxable amount is still showing incorrect. 

Are you using the tax_withholding feature for TCS and TDS or do you do it from the taxes and charges template?

You should be using the tax_witholding feature here.

**Comment by Govind-Jangid** [comment]
Yes, using Tax Withholding Category that TCS auto comes after 50Lakh sales.

File: india_compliance/gst_india/overrides/transaction.py [hunk 1]
```diff
@@ -46,15 +50,19 @@ def update_taxable_values(doc, valid_accounts):
...
                     and row.account_head in valid_accounts
                 ),
-                None,
+                None,  # ignore accounts after GST accounts
             )
 
         else:
+            # If no GST account is used
             reference_row_index = -1
+            # tax_witholding_amount = get_tds_amount(doc)
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/transaction.py]: uncomment

**Comment by Govind-Jangid** [comment]
@vorasmit
In GSTR1 showing duplicate rows, when TCS applied in Invoice.
image

**Comment by Govind-Jangid** [comment]
image

**Comment by vorasmit** [comment]
Thanks for reporting
fixed with #950

**Pull request merged**